src/main/java/org/mockito/internal/creation/settings/CreationSettings.java

```diff
@@ -104,6 +106,10 @@ public SerializableMode getSerializableMode() {
         return invocationListeners;
     }
 
+    public List<StubbingLookUpListener> getStubbingLookUpListeners() {
```

Comment:
Nice catch!

In short: Framework listener != Mock object listener.

Currently framework listeners are kept in thread-local state and hence they cannot receive events from different threads. It is legal to have the same mock shared in different threads for normal interactions with mock.

Mock object listeners are kept with the mock object, hence they can receive events from multiple threads.

I don't claim the current distinction is good :) It is best I could think of given current state of mockito architecture. I didn't want to have 'global' static listeners.

Instead of unifying the listeners, I tentatively plan to make the distinction explicit in the API, e.g. MockitoListener as top marker interface, and new MockObjectListener as another top marker interface for mock object listeners.